### PR TITLE
Use ES6 classes semantics

### PR DESCRIPTION
**Pull request by zbraniecki**

### CODE REVIEW

File: src/bindings/html/view.js [hunk 1]
```diff
@@ -13,37 +13,42 @@ const observerConfig = {
...
+  }
 
-View.prototype.formatEntity = function(id, args) {
-  return this.ctx.formatEntity(this.service.languages, id, args);
-};
+  formatEntity(id, args) {
+    return this.ctx.formatEntity(this.service.languages, id, args);
+  }
+
+  get(id) {
```
Comment (stasm) [inline src/bindings/html/view.js]: What's this? :)

File: src/lib/context.js [hunk 2]
```diff
@@ -4,133 +4,137 @@ import { L10nError } from './errors';
...
-  for (var key in entity.attrs) {
-    /* jshint -W089 */
-    var [attrLocals, attrValue] = this._formatTuple.call(
-      this, args, entity.attrs[key]);
-    formatted.attrs[key] = attrValue;
-    if (attrLocals.overlay) {
-      formatted.overlay = true;
+  /* private */
+
+  _formatTuple(args, entity) {
```
Comment (stasm) [inline src/lib/context.js:35]: Can we move the private methods out of the class definition?

Comment (stasm) [inline src/lib/context.js:35]: Hmm, maybe we shouldn't for now.  Resolver uses `_getEntity` and `_getMacro`.

File: src/lib/env.js [hunk 3]
```diff
@@ -13,80 +13,82 @@ const parsers = {
...
-
-  this._resMap = Object.create(null);
-  this._resCache = Object.create(null);
-
-  let listeners = {};
-  this.emit = emit.bind(this, listeners);
-  this.addEventListener = addEventListener.bind(this, listeners);
-  this.removeEventListener = removeEventListener.bind(this, listeners);
-}
+export default class Env{
```
Comment (stasm) [inline src/lib/env.js]: Nit: missing space before `{`.

File: src/lib/errors.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
 'use strict';
 
-export function L10nError(message, id, code) {
-  this.name = 'L10nError';
-  this.message = message;
-  this.id = id;
-  this.code = code;
+export class L10nError extends Error {
+  constructor(message, id, code) {
+    this.name = 'L10nError';
```
Comment (stasm) [inline src/lib/errors.js]: Did you try to build or run this? :)  You'll need to call `super` before you can use `this` in the constructor.

File: src/bindings/html/view.js (hunk 1 above)
Comment (zbraniecki) [inline src/bindings/html/view.js]: ah!

File: src/lib/errors.js (hunk 4 above)
Comment (zbraniecki) [inline src/lib/errors.js]: lint works, I built it using l20n-bundler and yes, I did run it.

The topic of super in constructor is kind of a heated one: https://esdiscuss.org/topic/super-on-class-that-extends - Gecko doesn't use it, babel requires it.

**Comment by zbraniecki** [comment]
updated the patch. it passes tests/lint.

**Comment by stasm** [comment]
No changes to L10nError?

**Comment by zbraniecki** [comment]
Spidermonkey errors when using super(), babel errors when super() is missing. We need both :) I'll leave L10nError until they figure out if super is mandatory or not and/or Gecko adds it.

**Comment by stasm** [comment]
Eh, I wish this was easier.  Thanks!

**Pull request merged**